tests/unit/useViolationsTest.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable rulesdir/prefer-underscore-method -- Disabled because it incorrectly flags the use of `it.each` */
+import { renderHook } from '@testing-library/react-native';
+import { useViolations, violationFields } from '@libs/Violations/useViolations';
+
+
+// Derive test cases from the violationFields object
+const allViolations = Object.keys(violationFields).map((name) => ({name}));
+// A list of all the field names that appear in `violationFields`
+const fieldNames = [...new Set(Object.values(violationFields))];
+// create an index of violations by field name
+const violationsByField = Object.entries(violationFields).reduce(
+    (acc, [violation, field]) => ({
+        ...acc,
+        [field]: [...(acc[field] || []), {name: violation}],
+    }),
+    {},
+);
+
+describe('useViolations', () => {
+    let violations = [];
+
+    beforeEach(() => {
+        violations = allViolations;
+    });
+
+    // The happy path
+    function callHook() {
+        const {result, rerender} = renderHook(() => useViolations(violations));
+        return {result, rerender};
+    }
+
+    it('returns correct values when there is only a single violation', () => {
+        const violation = violations[0];
+        violations = [violation];
+        const {result} = callHook();
+        const expectedField = violationFields[violation.name];
+        expect(result.current.hasViolations(expectedField)).toBe(true);
+        expect(result.current.getViolationsForField(expectedField)).toEqual([violation]);
+    });
+
+    it('returns correct values when there are multiple violations on different fields', () => {
+        const violation1 = violations[0];
+        const violation2 = violations[1];
+        violations = [violation1, violation2];
+
+        const {result} = callHook();
+
```

Comment:
```suggestion
        const {result} = callHook();
```

Reply:
Again this would fundamentally change the nature of the test, is that intended?

Reply:
I missed that it was necessary as you pointed out in the comment above. I think we can make a lot of these tests clearer by having these objects defined near the code making assertions